src/queue.js

```diff
@@ -0,0 +1,35 @@
+'use strict'
+
+export default class Queue {
+  constructor() {
+    this.storage = [],
+    this.head = 0,
```

Comment:
Nitpick: queue naming convention would be `this.front` and `this.back` instead of `this.head` and `this.end`.

Changing the way your `Queue` internals are named will also add consistency to your code.